test/e2e/flowexporter_test.go

```diff
@@ -175,60 +175,61 @@ func checkRecordsWithPodIPs(t *testing.T, data *TestData, podAIP string, podBIP
-			if strings.Contains(record, "seqno=2") || strings.Contains(record, "seqno=3") {
 				// In Kind clusters, there are two flow records for the iperf flow.
 				// One of them has no bytes and we ignore that flow record.
 				if !strings.Contains(record, "octetDeltaCount: 0") {
-					//split the record in lines to compute bandwidth
-					splitLines := strings.Split(record, "\n")
-					for _, line := range splitLines {
-						if strings.Contains(line, "octetDeltaCount") {
-							lineSlice := strings.Split(line, ":")
-							deltaBytes, err := strconv.ParseFloat(strings.TrimSpace(lineSlice[1]), 64)
-							if err != nil {
-								t.Fatalf("Error in converting octetDeltaCount to int type")
-							}
-							// compute the bandwidth using 5s as interval
-							recBandwidth := (deltaBytes * 8.0) / float64(5*time.Second.Nanoseconds())
-							// bandwidth from iperf output
-							bwSlice := strings.Split(bandwidth, " ")
-							iperfBandwidth, err := strconv.ParseFloat(bwSlice[0], 64)
-							if err != nil {
-								t.Fatalf("Error in converting iperf bandwidth to float64 type")
+					// Check if records have both ingress and egress network policies.
+					if !strings.Contains(record, "test-networkpolicy-ingress") {
+						t.Fatalf("Records does not have NetworkPolicy name with ingress rule")
+					}
+					if !strings.Contains(record, "test-networkpolicy-egress") {
+						t.Fatalf("Records does not have NetworkPolicy name with egress rule")
+					}
+				}
+				// Check the bandwidth using octetDeltaCount in data records sent in second ipfix interval
+				if strings.Contains(record, "seqno=2") || strings.Contains(record, "seqno=3") {
+					// In Kind clusters, there are two flow records for the iperf flow.
+					// One of them has no bytes and we ignore that flow record.
+					if !strings.Contains(record, "octetDeltaCount: 0") {
+						//split the record in lines to compute bandwidth
+						splitLines := strings.Split(record, "\n")
+						for _, line := range splitLines {
+							if strings.Contains(line, "octetDeltaCount") {
+								lineSlice := strings.Split(line, ":")
+								deltaBytes, err := strconv.ParseFloat(strings.TrimSpace(lineSlice[1]), 64)
+								if err != nil {
+									t.Fatalf("Error in converting octetDeltaCount to int type")
+								}
+								// compute the bandwidth using 5s as interval
+								recBandwidth := (deltaBytes * 8.0) / float64(5*time.Second.Nanoseconds())
+								// bandwidth from iperf output
+								bwSlice := strings.Split(bandwidth, " ")
+								iperfBandwidth, err := strconv.ParseFloat(bwSlice[0], 64)
+								if err != nil {
+									t.Fatalf("Error in converting iperf bandwidth to float64 type")
+								}
+								t.Logf("Iperf bandwidth: %v", iperfBandwidth)
+								t.Logf("IPFIX record bandwidth: %v", recBandwidth)
+								assert.InEpsilonf(t, recBandwidth, iperfBandwidth, 5, "Difference between Iperf bandwidth and IPFIX record bandwidth should be less than 5Gb/s")
```

Comment:
Not related to your PR. I made an error here when using `assert.InEpsilonf`. Could you use `assert.InDeltaf` instead?

Reply:
Sure. updated